File: Java/백준/Silver/2841. 외계인의 기타 연주/외계인의 기타 연주.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        List<Stack<Integer>> lists = new ArrayList<>();
        for (int i = 0; i < 6; i++) {
            lists.add(new Stack<>());
        }

        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int p = Integer.parseInt(st.nextToken());

        int count = 0;

        for (int i = 0; i < n; i++) {
            StringTokenizer st1 = new StringTokenizer(br.readLine());
            int line = Integer.parseInt(st1.nextToken());
            int num = Integer.parseInt(st1.nextToken());
            line -= 1;

            while (!lists.get(line).isEmpty() && lists.get(line).peek() > num) {
                lists.get(line).pop();
                count++;
            }

            if (!lists.get(line).isEmpty() && lists.get(line).peek() == num) {
                continue;
            }

            lists.get(line).push(num);
            count++;
        }
        System.out.println(count);
    }
}

File: Python/백준/Silver/2841. 외계인의 기타 연주/외계인의 기타 연주.py
import sys
input = sys.stdin.readline

arr = [[] for _ in range(6)]
n, p = map(int, input().split())
count = 0

for _ in range(n):
    line, num = map(int, input().split())
    line -= 1

    while arr[line] and arr[line][-1] > num:
        arr[line].pop()
        count += 1

    if arr[line] and arr[line][-1] == num:
        continue

    arr[line].append(num)
    count += 1

print(count)